src/lib/Libnet/net_server.c

```diff
@@ -520,14 +556,42 @@ wait_request(time_t waittime)
 	nfds = tpp_em_wait(poll_context, &events, timeout);
 	err = errno;
 #endif /* WIN32 */
-
 	if (nfds < 0) {
 		if (!(err == EINTR || err == EAGAIN || err == 0)) {
 			snprintf(logbuf, sizeof(logbuf), " tpp_em_wait() error, errno=%d", err);
 			log_err(err, __func__, logbuf);
 			return (-1);
 		}
 	} else {
+		prio_sock_processed = 0;
+		if (priority_context) {
+			em_event_t *pevents;
+			timeout = 0;
+#ifndef WIN32
+        		/* wait after unblocking signals in an atomic call */
+        		sigemptyset(&emptyset);
+        		pnfds = tpp_em_pwait(priority_context, &pevents, timeout, &emptyset);
+        		err = errno;
+
+#else
+        		errno = 0;
+        		pnfds = tpp_em_wait(priority_context, &pevents, timeout);
+        		err = errno;
+#endif /* WIN32 */
+                	for (i = 0; i < pnfds; i++) {
+                        	int em_pfd;
```

Comment:
done

Reply:
I'm not sure I agree.  Unless there is a reason to do otherwise, I think variables should be as tightly scoped as possible.  There won't be a runtime difference since variable declarations will be handled at compile time.